src/org/openlcb/cdi/jdom/CdiMemConfigReader.java

```diff
@@ -105,9 +106,18 @@ public void handleReadData(NodeID dest, int space, long address, byte[] data) {
     private void done() {
         // done, pass back a reader based on the current buffer contents
         if (retval != null) {
-            retval.progressNotify(buf.length(), buf.length());
-            logger.log(Level.FINE, "Retrieved XML: \n{0}", buf);
-            retval.provideReader(new java.io.StringReader(new String(buf)));
+            retval.progressNotify(buf.size(), buf.size());
+            byte[] byteArray = new byte[buf.size()];
+            for (int i = 0 ; i<buf.size(); i++) {
+                byteArray[i] = (byte)buf.get(i);
+            }
+            try {
+                String xml = new String(byteArray, "UTF-8");
+                logger.log(Level.FINE, "Retrieved XML: \n{0}", xml);
+                retval.provideReader(new java.io.StringReader(xml));
+            } catch (java.io.UnsupportedEncodingException e) {
+                logger.warning("UnsupportedEncodingException while preparing XML data");
```

Comment:
I think we should provide some string back even in the error case. It could be an empty string. Otherwise the UI will get stuck.

Reply:
This is one of those "can't realistically happen, but has to be caught" exceptions.  It'll only fire if the `new String(byteArray, "UTF-8")` expression decides that "UTF-8" is not a valid encoding.  If that happens, there's something broken deep within Java.

That said, we could still provide something in `retval`.  I'm just not sure what that content should be.  An empty string?  A minimal CDI with no content inside the `<cdi>` element?

Reply:
ok i misunderstood and thought that broken input utf8 tokens will also cause this error path.
Maybe we should just have a finally { provide(new StringREader("")) }

Reply:
actually finally {} is a bad idea. I take that back.

But providing an empty string in the error path is reasonable.